deploy/kubernetes-1.18/wekafs/csi-wekafs-plugin.yaml

```diff
@@ -65,106 +66,121 @@ rules:
 metadata:
-  name: csi-wekafsplugin-cluster-role-binding
+  name: csi-wekafsplugin-controller
 subjects:
   - kind: ServiceAccount
-    name: csi-wekafsplugin
+    name: csi-wekafsplugin-controller
     namespace: csi-wekafsplugin
 roleRef:
   kind: ClusterRole
-  name: csi-wekafsplugin-cluster-role
+  name: csi-wekafsplugin-controller
   apiGroup: rbac.authorization.k8s.io
 
-# role
 ---
-kind: Role
-apiVersion: rbac.authorization.k8s.io/v1
-metadata:
-  namespace: csi-wekafsplugin
-  name: csi-wekafsplugin-role
-rules:
-  - apiGroups: ["coordination.k8s.io"]
-    resources: ["leases"]
-    verbs: ["get", "watch", "list", "delete", "update", "create"]
-  - apiGroups: [""]
-    resources: ["endpoints"]
-    verbs: ["get", "watch", "list", "delete", "update", "create"]
-
-# role binding
----
-kind: RoleBinding
+kind: ClusterRoleBinding
 apiVersion: rbac.authorization.k8s.io/v1
 metadata:
-  name: csi-wekafsplugin-role-binding
-  namespace: csi-wekafsplugin
+  name: csi-wekafsplugin-node
 subjects:
   - kind: ServiceAccount
-    name: csi-wekafsplugin
+    name: csi-wekafsplugin-node
     namespace: csi-wekafsplugin
 roleRef:
-  kind: Role
-  name: csi-wekafsplugin-role
+  kind: ClusterRole
+  name: csi-wekafsplugin-node
   apiGroup: rbac.authorization.k8s.io
 
-# plugin daemonset definition
+
+# controller pod StatefulSet definition
 ---
-kind: DaemonSet
+kind: StatefulSet
 apiVersion: apps/v1
 metadata:
-  name: csi-wekafsplugin
+  name: csi-wekafsplugin-controller
   namespace: csi-wekafsplugin
 spec:
+  serviceName: csi-wekafsplugin-controller
   selector:
     matchLabels:
       kubernetes.io/os: linux
+  replicas: 1
   template:
     metadata:
       labels:
         app: csi-wekafsplugin
         kubernetes.io/os: linux
     spec:
-      serviceAccountName: csi-wekafsplugin
+      serviceAccountName: csi-wekafsplugin-controller
       containers:
-        - name: node-driver-registrar
-          image: quay.io/k8scsi/csi-node-driver-registrar:v1.3.0
-          args:
-            - --v=5
-            - --csi-address=/csi/csi.sock
-            - --kubelet-registration-path=/var/lib/kubelet/plugins/csi-wekafs/csi.sock
-          securityContext:
-            # This is necessary only for systems with SELinux, where
-            # non-privileged sidecar containers cannot access unix domain socket
-            # created by privileged CSI driver container.
-            privileged: true
-          env:
-            - name: KUBE_NODE_NAME
-              valueFrom:
-                fieldRef:
-                  apiVersion: v1
-                  fieldPath: spec.nodeName
-          volumeMounts:
-          - mountPath: /csi
-            name: socket-dir
-          - mountPath: /registration
-            name: registration-dir
-          - mountPath: /var/lib/csi-wekafs-data
-            name: csi-data-dir
-
         - name: wekafs
-          image: quay.io/weka.io/csi-wekafs:latest
+          image: quay.io/weka.io/csi-wekafs:v0.6.2-0
```

Comment:
why hard coded? It should come out of helm version/values

Reply:
The csi-wekafs-plugin.yaml is used with legacy deploy.sh installation, not through helm.